doc/mgmt/gnmi/SONiC_GNOI_Server_Interface_Design.md

```diff
@@ -0,0 +1,653 @@
+# SONiC GNOI Server Interface Design
+
+# High Level Design Document
+
+#### Rev 0.1
+
+# Table of Contents
+- [Table of Contents](#table-of-contents)
+- [Revision](#revision)
+- [About this Document](#about-this-document)
+- [Definition/Abbreviation](#definitionabbreviation)
+- [Background](#background)
+- [Scope](#scope)
+- [1 Project Goal Summary](#1-project-goal-summary)
+- [2 Requirements](#2-requirements)
+- [3 Architecture Design](#3-architecture-design)
+- [4 High Level Design](#4-high-level-design)
+- [5 Sample Scenario](#5-sample-scenario)
+- [6 Testing](#6-testing)
+
+
+
+# Revision
+
+| Rev | Date        | Author             | Change Description  |
+|:---:|:-----------:|:------------------:|---------------------|
+| 0.1  | 03/27/2024 | Isabel Li    | Initial version     |
+
+# About this Document
+This document provides a detailed description on the strategy to implement the SONiC GNOI Server Interface.
+
+# Definition/Abbreviation
+| **Term** | **Meaning**                |
+| -------- | -------------------------- |
+| API      | Application Programming Interface |
+| CLI      | Command Line Interface |
+| gRPC     | A modern open-source high performance Remote Procedure Call (RPC) framework that can run in any environment |
+| gNMI     | gRPC Network Management Interface, used to retrieve or manipulate the state of a device via telemetry or configuration data |
+| gNOI     | gRPC Network Operations Interface |
+| NE       | Network Engineers |
+
+# Background
+Software for Open Networking in the Cloud (SONiC) is an open source network operating system (NOS) based on Linux that runs on switches from multiple vendors and ASICs. SONiC offers a full suite of network functionality, like BGP and RDMA, that has been production-hardened in the data centers of some of the largest cloud service providers. 
+
+Currently, most SONiC operation and configuration management requires directly using CLI on the device. This document proposes the addition of gNOI API to SONiC to modernize SONiC operation management. 
+
+GNOI (gRPC network operations interface) is supported through OpenConfig. All gNOI services and documentation are available on the [OpenConfig page](https://github.com/openconfig/gnoi). OpenConfig defines and implements a common, vendor-independent software layer for managing network devices.
```

Comment:
Please capture the steps/guidelines to extend the standard gnoi proto files for new fields if needed.

Reply:
We don't currently have a plan to extend standard gNOI proto files for new fields. If the API needed is not already supported by gNOI or SONiC internal APIs, we have the following two options: 

1. Add new internal SONiC API (ref: https://github.com/sonic-net/sonic-gnmi/blob/master/proto/gnoi/sonic_gnoi.proto)
2. Propose update to gNOI protocol